### PR TITLE
Auto Bundles were not working in AMD Environment

### PR DESCRIPTION
**Pull request by azizhk**
This was because of this code:
https://github.com/stefanpenner/es6-promise/blob/f5f44e81e3d92db1445d61c157a3593a3662ef81/dist/es6-promise.auto.js#L11-L12

`ES6Promise` would not be defined on global and we would get an error saying, "Cannot read property polyfill of undefined".
Though there are multiple workarounds, I thought this would be the best way to fix this.

I would have preferred to export nothing, just call the polyfill function, which would make it inline in the minified version, but that would break semver.
Also I would love to change the module name from ES6Promise to es6-promise, to keep it the same in NPM and on the browser. (But that would make it `window['es6-promise']` for non AMD 😢 )
Also this is my first time with `Broccoli.js`. So please correct me if I'm doing something wrong.
Also I did not commit all of the dist files. Let me know if I should not commit them.

### CODE REVIEW

**Comment by stefanpenner** [comment]
@azizhk thanks for digging into this. Do you think we should just always install the ES6Promise global even in AMD environments? I suspect the lack of this may be causing https://github.com/stefanpenner/es6-promise/issues/257 as well, as causing this issue.

What do you think?

**Comment by azizhk** [comment]
> Do you think we should just always install the ES6Promise global even in AMD environments?
If you mean we should just set window.ES6Promise then no, we should not.
If you mean we should polyfill it then yes. He is requesting for the auto version.

**Comment by stefanpenner** [comment]
> If you mean we should just set window.ES6Promise then no, we should not.

👍 

> If you mean we should polyfill it then yes. He is requesting for the auto version.

Good catch.

**Comment by stefanpenner** [comment]
I'm trying to replicate the above test failure locally, but without much luck...

**Comment by azizhk** [comment]
Same here. I was pretty sure they were working when I opened my PR. Damn.

**Comment by stefanpenner** [comment]
I'm rebuilding master on travis, lets see if its node 4 also fails. If so, then maybe its some transitive?

**Comment by stefanpenner** [comment]
Ya, it failed again. I really wish it wasn't so cryptic...

File: Brocfile.js [hunk 1]
```diff
@@ -33,20 +33,26 @@ var es5 = new Babel(lib, {
   blacklist: ['es6.modules']
 });
 
+function rollupConfig(entry) {
+  return new Rollup(es5, {
+    rollup: {
+      entry: 'lib/' + entry,
+      cache: 'BUILD:es6-promise',
```
Comment (stefanpenner) [inline Brocfile.js]: What does naming it do?

Comment (azizhk) [inline Brocfile.js]: We have two rollups with almost every file repeated so to reuse the same cache (of file reads, AST etc.) for both of them.
https://github.com/rollup/rollup/wiki/JavaScript-API

Let me find a better Doc

Comment (azizhk) [inline Brocfile.js]: ok I'm not doing it correctly.

Comment (azizhk) [inline Brocfile.js]: https://github.com/chadhietala/broccoli-rollup/blob/master/src/index.js
Ok just saw this plugin automatically does it for me. I was surprised why was I getting good speeds, the plugin was overriding the cache variable for me. 😜

Comment (azizhk) [inline Brocfile.js]: https://github.com/chadhietala/broccoli-rollup/commit/5aaf8f38f723a3d6e0f6160fa65888ed65c0329d
That was added by you. I feel like such an idiot.

Comment (stefanpenner) [inline Brocfile.js]: @azizhk haha no worries, I thought maybe you found some better caching strategy hidden deep inside rollup itself.

**Comment by stefanpenner** [comment]
Im really not sure whats up. I have tested this thoroughly locally in node 4 without any issues.. Going to merge this.

**Comment by stefanpenner** [comment]
moving node 4 to allowed failures for now: https://github.com/stefanpenner/es6-promise/pull/272

Will test node 4 locally to confirm until we resolved the node 4 issue.

**Pull request merged**